projectq/ops/_basics.py

```diff
@@ -295,23 +295,29 @@ def get_merged(self, other):
 
     def __eq__(self, other):
         """ Return True if same class and same rotation angle. """
-        tolerance = 1.e-12
-        if isinstance(other, self.__class__):
-            difference = abs(self._angle - other._angle)
```

Comment:
Yes, I would keep the double comparison instead of introducing another function call. But I would take your suggestions that it would be better `difference = ... % (4 * math.pi)` as with the modulo operation we don't rely on the angle being within [0,4*math.pi) but that it works generally. Also a small simplification of the original code can be done by removing the last two
```
else: 
   return False
else:
   return False
```
to just
`return False` if none of the above two `if` were true.